masonite/helpers/routes.py

```diff
@@ -178,8 +178,14 @@ def create_matchurl(router, route):
     regex = route.compile_route_to_regex(router)
 
     if route.route_url.endswith('/'):
-        matchurl = re.compile(regex.replace(r'\/\/$', r'\/$'))
+        if router.url.endswith('/'):
+            matchurl = re.compile(regex.replace(r'\/\/$', r'\/$'))
+        else:
+            matchurl = re.compile(regex.replace(r'\/\/$', r'$'))
     else:
-        matchurl = re.compile(regex.replace(r'\/$', r'$'))
+        if router.url.endswith('/'):
+            matchurl = re.compile(regex)
+        else:
+            matchurl = re.compile(regex.replace(r'\/$', r'$'))
 
     return matchurl
```

Comment:
🤔 I'll have to play with this when I pull it in. You might be able to condense this logic a bit.

Sort of a note to myself when I test this